flagd/pkg/service/flag-evaluation/connect_service.go

```diff
@@ -183,14 +183,16 @@ func (s *ConnectService) setupServer(svcConf service.Configuration) (net.Listene
 	s.serverMtx.Unlock()
 
 	// Add middlewares
-	metricsMiddleware := metricsmw.NewHTTPMetric(metricsmw.Config{
-		Service:        svcConf.ServiceName,
-		MetricRecorder: s.metrics,
-		Logger:         s.logger,
-		HandlerID:      "",
-	})
-
-	s.AddMiddleware(metricsMiddleware)
+	if s.metrics != nil {
```

Comment:
should we fallback to`NoopMetricsRecorder`  so we can avoid this check 🤔

Reply:
ah good catch, thanks - was introducing the NoopMetricsProvider to avoid these kinds of checks, but seems like i forgot to remove this here :)